packages/envied_generator/lib/src/generate_field.dart

```diff
@@ -10,11 +10,49 @@ import 'package:source_gen/source_gen.dart';
 /// Since this function also does the type casting,
 /// an [InvalidGenerationSourceError] will also be thrown if
 /// the type can't be casted, or is not supported.
-Iterable<Field> generateFields(FieldElement field, String value) {
+Iterable<Field> generateFields(
+  FieldElement field,
+  String? value, {
+  required bool optional,
+}) {
   final String type = field.type.getDisplayString(withNullability: false);
 
   late final Expression result;
 
+  if (optional && value == null) {
+    // Early return if null, so need to check for allowed types
+    if (!field.type.isDartCoreInt &&
+        !field.type.isDartCoreDouble &&
+        !field.type.isDartCoreNum &&
+        !field.type.isDartCoreBool &&
+        !field.type.isDartCoreString &&
+        field.type is! DynamicType) {
+      throw InvalidGenerationSourceError(
+        'Envied can only handle types such as `int`, `double`, `num`, `bool` and'
+        ' `String`. Type `$type` is not one of them.',
+        element: field,
+      );
+    }
+
+    return [
+      Field(
+        (FieldBuilder fieldBuilder) => fieldBuilder
+          ..static = true
+          ..modifier = FieldModifier.constant
+          ..type = refer(field.type is DynamicType
+              ? ''
+              : field.type.getDisplayString(withNullability: true))
+          ..name = field.name
+          ..assignment = Code('null'),
+      ),
+    ];
+  } else if (value == null) {
```

Comment:
You can remove the `else` here.

Reply:
Did you mean to remove the `else` block or just remove the `else` and put the `if` on a separate line?

I did the first and fixed the failing Env2 `dynamic` test that was hidden by this.

Reply:
Unresolve this if thats not what you wanted

Reply:
I wanted you to replace `} else if () {` with simply `if () {}` since you are using `return` in both instances.

Reply:
Alright, I will add that part back.

Reply:
Gotcha! I also renamed two variable names since it was a little confusing in `generateFieldsEncrypted`